e2e/terraform/shared/config/provision-client.sh

```diff
@@ -22,11 +22,26 @@ sleep 10
 # Add hostname to /etc/hosts
 echo "127.0.0.1 $(hostname)" | sudo tee --append /etc/hosts
 
-# Add Docker bridge network IP to /etc/resolv.conf (at the top)
+# Use dnsmasq first and then docker bridge network for DNS resolution
 DOCKER_BRIDGE_IP_ADDRESS=$(/usr/local/bin/sockaddr eval 'GetInterfaceIP "docker0"')
-echo "nameserver $DOCKER_BRIDGE_IP_ADDRESS" | sudo tee /etc/resolv.conf.new
-cat /etc/resolv.conf | sudo tee --append /etc/resolv.conf.new
-sudo mv /etc/resolv.conf.new /etc/resolv.conf
+{
+    echo "nameserver 127.0.0.1"
+    echo "nameserver $DOCKER_BRIDGE_IP_ADDRESS"
+} > /tmp/resolv.conf
+sudo mv /tmp/resolv.conf /etc/resolv.conf
+
+# need to get the AWS DNS address from the VPC...
+# this is pretty hacky but will work for any typical case
+MAC=$(curl -s --fail http://169.254.169.254/latest/meta-data/mac)
+CIDR_BLOCK=$(curl -s --fail "http://169.254.169.254/latest/meta-data/network/interfaces/macs/$MAC/vpc-ipv4-cidr-block")
+VPC_DNS_ROOT=$(echo "$CIDR_BLOCK" | cut -d'.' -f1-3)
+{
+    echo "nameserver ${VPC_DNS_ROOT}.2"
+} > /tmp/dnsmasq-resolv.conf
```

Comment:
If it's one line, we probably don't need the quotation.

Reply:
Oops, yup... forgot to fix that when I removed the 8.8.8.8 nameserver from an earlier iteration. Done.